### ISSUE TITLE
Can't use `print-identity-token` with authenticated service account from workload identity pool.

### ISSUE THREAD

**Issue by di** [google-github-actions/auth#291]
### TL;DR

I tried the workflow at https://github.com/google-github-actions/setup-gcloud#usage and couldn't run `gcloud auth print-identity-token` with the result.

### Expected behavior

I expected to be able to print an identity token for the authenticated service account, and also use this to do things like `gcloud functions call` (which also fails).

### Observed behavior

The service account is successfully authenticated, but I get an error:

```
ERROR: (gcloud.auth.print-identity-token) No identity token can be obtained from the current credentials.
```

### Action YAML

```yaml
name: GitHub Actions Demo
on: [push]
jobs:
  job_id:
    runs-on: ubuntu-latest
    # Add "id-token" with the intended permissions.
    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
    # This is required to share credentials between steps
    - uses: 'actions/checkout@v3'
    
    - id: 'auth'
      uses: 'google-github-actions/auth@v1'
      with:
        workload_identity_provider: 'projects/988791436232/locations/global/workloadIdentityPools/conformance-testing-id-pool/providers/github-actions'
        service_account: 'conformance-testing-invoker@sigstore-test-364020.iam.gserviceaccount.com'

    - name: 'Set up Cloud SDK'
      uses: 'google-github-actions/setup-gcloud@v1'
      with:
        version: '>= 363.0.0'

    - name: 'Use gcloud CLI'
      run: 'gcloud auth print-identity-token'
```

### Log output

```text
2023-04-04T01:42:37.7687227Z Requested labels: ubuntu-latest
2023-04-04T01:42:37.7687338Z Job defined at: di/identity-token-test/.github/workflows/test.yml@refs/heads/main
2023-04-04T01:42:37.7687410Z Waiting for a runner to pick up this job...
2023-04-04T01:42:38.4155751Z Job is waiting for a hosted runner to come online.
2023-04-04T01:42:41.3457272Z Job is about to start running on the hosted runner: GitHub Actions 2 (hosted)
2023-04-04T01:42:47.0186620Z Current runner version: '2.303.0'
2023-04-04T01:42:47.0222470Z ##[group]Operating System
2023-04-04T01:42:47.0223242Z Ubuntu
2023-04-04T01:42:47.0223580Z 22.04.2
2023-04-04T01:42:47.0223832Z LTS
2023-04-04T01:42:47.0224321Z ##[endgroup]
2023-04-04T01:42:47.0224716Z ##[group]Runner Image
2023-04-04T01:42:47.0225146Z Image: ubuntu-22.04
2023-04-04T01:42:47.0225531Z Version: 20230326.2
2023-04-04T01:42:47.0226138Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20230326.2/images/linux/Ubuntu2204-Readme.md
2023-04-04T01:42:47.0226861Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230326.2
2023-04-04T01:42:47.0227396Z ##[endgroup]
2023-04-04T01:42:47.0227852Z ##[group]Runner Image Provisioner
2023-04-04T01:42:47.0228199Z 2.0.139.1
2023-04-04T01:42:47.0228654Z ##[endgroup]
2023-04-04T01:42:47.0229418Z ##[group]GITHUB_TOKEN Permissions
2023-04-04T01:42:47.0230157Z Contents: read
2023-04-04T01:42:47.0230480Z Metadata: read
2023-04-04T01:42:47.0231221Z ##[endgroup]
2023-04-04T01:42:47.0235856Z Secret source: Actions
2023-04-04T01:42:47.0236917Z Prepare workflow directory
2023-04-04T01:42:47.1170686Z Prepare all required actions
2023-04-04T01:42:47.1411409Z Getting action download info
2023-04-04T01:42:47.5142531Z Download action repository 'actions/checkout@v3' (SHA:8f4b7f84864484a7bf31766abe9204da3cbe65b3)
2023-04-04T01:42:48.1756082Z Download action repository 'google-github-actions/auth@v1' (SHA:ef5d53e30bbcd8d0836f4288f5e50ff3e086997d)
2023-04-04T01:42:48.6266593Z Download action repository 'google-github-actions/setup-gcloud@v1' (SHA:62d4898025f6041e16b1068643bfc5a696863587)
2023-04-04T01:42:49.2296992Z Complete job name: job_id
2023-04-04T01:42:49.3450061Z ##[group]Run actions/checkout@v3
2023-04-04T01:42:49.3450411Z with:
2023-04-04T01:42:49.3450681Z   repository: di/identity-token-test
2023-04-04T01:42:49.3451217Z   token: ***
2023-04-04T01:42:49.3451467Z   ssh-strict: true
2023-04-04T01:42:49.3451718Z   persist-credentials: true
2023-04-04T01:42:49.3451997Z   clean: true
2023-04-04T01:42:49.3452448Z   fetch-depth: 1
2023-04-04T01:42:49.3452740Z   lfs: false
2023-04-04T01:42:49.3452960Z   submodules: false
2023-04-04T01:42:49.3453213Z   set-safe-directory: true
2023-04-04T01:42:49.3453459Z ##[endgroup]
2023-04-04T01:42:49.7421256Z Syncing repository: di/identity-token-test
2023-04-04T01:42:49.7423233Z ##[group]Getting Git version info
2023-04-04T01:42:49.7423820Z Working directory is '/home/runner/work/identity-token-test/identity-token-test'
2023-04-04T01:42:49.7424433Z [command]/usr/bin/git version
2023-04-04T01:42:49.7539669Z git version 2.40.0
2023-04-04T01:42:49.7581747Z ##[endgroup]
2023-04-04T01:42:49.7625122Z Temporarily overriding HOME='/home/runner/work/_temp/2f0a6e2f-9b77-4258-9a62-6d723e8c423e' before making global git config changes
2023-04-04T01:42:49.7625805Z Adding repository directory to the temporary git global config as a safe directory
2023-04-04T01:42:49.7633469Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/identity-token-test/identity-token-test
2023-04-04T01:42:49.7704083Z Deleting the contents of '/home/runner/work/identity-token-test/identity-token-test'
2023-04-04T01:42:49.7711709Z ##[group]Initializing the repository
2023-04-04T01:42:49.7717646Z [command]/usr/bin/git init /home/runner/work/identity-token-test/identity-token-test
2023-04-04T01:42:49.7803061Z hint: Using 'master' as the name for the initial branch. This default branch name
2023-04-04T01:42:49.7803774Z hint: is subject to change. To configure the initial branch name to use in all
2023-04-04T01:42:49.7805650Z hint: of your new repositories, which will suppress this warning, call:
2023-04-04T01:42:49.7805977Z hint: 
2023-04-04T01:42:49.7806392Z hint: 	git config --global init.defaultBranch <name>
2023-04-04T01:42:49.7806652Z hint: 
2023-04-04T01:42:49.7807028Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2023-04-04T01:42:49.7807638Z hint: 'development'. The just-created branch can be renamed via this command:
2023-04-04T01:42:49.7808312Z hint: 
2023-04-04T01:42:49.7808614Z hint: 	git branch -m <name>
2023-04-04T01:42:49.7824777Z Initialized empty Git repository in /home/runner/work/identity-token-test/identity-token-test/.git/
2023-04-04T01:42:49.7843172Z [command]/usr/bin/git remote add origin https://github.com/di/identity-token-test
2023-04-04T01:42:49.7899516Z ##[endgroup]
2023-04-04T01:42:49.7900464Z ##[group]Disabling automatic garbage collection
2023-04-04T01:42:49.7905413Z [command]/usr/bin/git config --local gc.auto 0
2023-04-04T01:42:49.7944446Z ##[endgroup]
2023-04-04T01:42:49.7945078Z ##[group]Setting up auth
2023-04-04T01:42:49.7954053Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-04-04T01:42:49.7995812Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-04-04T01:42:49.8433711Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-04-04T01:42:49.8510219Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-04-04T01:42:49.8784735Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2023-04-04T01:42:49.8826750Z ##[endgroup]
2023-04-04T01:42:49.8827264Z ##[group]Fetching the repository
2023-04-04T01:42:49.8842238Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +e5ff4f67befd106d5f9f3742bf935b46ac24dd7b:refs/remotes/origin/main
2023-04-04T01:42:50.2558146Z remote: Enumerating objects: 5, done.        
2023-04-04T01:42:50.2558520Z remote: Counting objects:  20% (1/5)        
2023-04-04T01:42:50.2558829Z remote: Counting objects:  40% (2/5)        
2023-04-04T01:42:50.2559108Z remote: Counting objects:  60% (3/5)        
2023-04-04T01:42:50.2559428Z remote: Counting objects:  80% (4/5)        
2023-04-04T01:42:50.2559717Z remote: Counting objects: 100% (5/5)        
2023-04-04T01:42:50.2560004Z remote: Counting objects: 100% (5/5), done.        
2023-04-04T01:42:50.2560318Z remote: Compressing objects:  50% (1/2)        
2023-04-04T01:42:50.2560917Z remote: Compressing objects: 100% (2/2)        
2023-04-04T01:42:50.2561246Z remote: Compressing objects: 100% (2/2), done.        
2023-04-04T01:42:50.2561878Z remote: Total 5 (delta 0), reused 0 (delta 0), pack-reused 0        
2023-04-04T01:42:50.2624203Z From https://github.com/di/identity-token-test
2023-04-04T01:42:50.2628717Z  * [new ref]         e5ff4f67befd106d5f9f3742bf935b46ac24dd7b -> origin/main
2023-04-04T01:42:50.2656460Z ##[endgroup]
2023-04-04T01:42:50.2657208Z ##[group]Determining the checkout info
2023-04-04T01:42:50.2659386Z ##[endgroup]
2023-04-04T01:42:50.2660202Z ##[group]Checking out the ref
2023-04-04T01:42:50.2665987Z [command]/usr/bin/git checkout --progress --force -B main refs/remotes/origin/main
2023-04-04T01:42:50.2718712Z Switched to a new branch 'main'
2023-04-04T01:42:50.2723983Z branch 'main' set up to track 'origin/main'.
2023-04-04T01:42:50.2729253Z ##[endgroup]
2023-04-04T01:42:50.2776723Z [command]/usr/bin/git log -1 --format='%H'
2023-04-04T01:42:50.2809507Z 'e5ff4f67befd106d5f9f3742bf935b46ac24dd7b'
2023-04-04T01:42:50.3262541Z ##[group]Run google-github-actions/auth@v1
2023-04-04T01:42:50.3262903Z with:
2023-04-04T01:42:50.3263420Z   workload_identity_provider: projects/988791436232/locations/global/workloadIdentityPools/conformance-testing-id-pool/providers/github-actions
2023-04-04T01:42:50.3264176Z   service_account: conformance-testing-invoker@sigstore-test-364020.iam.gserviceaccount.com
2023-04-04T01:42:50.3264675Z   create_credentials_file: true
2023-04-04T01:42:50.3264982Z   export_environment_variables: true
2023-04-04T01:42:50.3265336Z   cleanup_credentials: true
2023-04-04T01:42:50.3265722Z   access_token_lifetime: 3600s
2023-04-04T01:42:50.3266398Z   access_token_scopes: https://www.googleapis.com/auth/cloud-platform
2023-04-04T01:42:50.3266795Z   retries: 0
2023-04-04T01:42:50.3267107Z   id_token_include_email: false
2023-04-04T01:42:50.3267368Z ##[endgroup]
2023-04-04T01:42:50.5746534Z Created credentials file at "/home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json"
2023-04-04T01:42:50.5911515Z ##[group]Run google-github-actions/setup-gcloud@v1
2023-04-04T01:42:50.5911853Z with:
2023-04-04T01:42:50.5912182Z   version: >= 363.0.0
2023-04-04T01:42:50.5912464Z   skip_install: false
2023-04-04T01:42:50.5912888Z env:
2023-04-04T01:42:50.5913307Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json
2023-04-04T01:42:50.5913977Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json
2023-04-04T01:42:50.5914579Z   GOOGLE_GHA_CREDS_PATH: /home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json
2023-04-04T01:42:50.5915092Z   CLOUDSDK_CORE_PROJECT: sigstore-test-364020
2023-04-04T01:42:50.5915465Z   CLOUDSDK_PROJECT: sigstore-test-364020
2023-04-04T01:42:50.5915872Z   GCLOUD_PROJECT: sigstore-test-364020
2023-04-04T01:42:50.5916491Z   GCP_PROJECT: sigstore-test-364020
2023-04-04T01:42:50.5917002Z   GOOGLE_CLOUD_PROJECT: sigstore-test-364020
2023-04-04T01:42:50.5917282Z ##[endgroup]
2023-04-04T01:42:52.3967406Z [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/f5613133-b62a-425d-8c47-0da1800d115b -f /home/runner/work/_temp/f8b3d2b6-780c-4dc4-b0ca-d9862b15ddcc
2023-04-04T01:43:17.8781878Z Successfully authenticated
2023-04-04T01:43:17.8998940Z ##[group]Run gcloud auth print-identity-token
2023-04-04T01:43:17.8999391Z [36;1mgcloud auth print-identity-token[0m
2023-04-04T01:43:17.9061040Z shell: /usr/bin/bash -e {0}
2023-04-04T01:43:17.9061275Z env:
2023-04-04T01:43:17.9063190Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json
2023-04-04T01:43:17.9063839Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json
2023-04-04T01:43:17.9064403Z   GOOGLE_GHA_CREDS_PATH: /home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json
2023-04-04T01:43:17.9064820Z   CLOUDSDK_CORE_PROJECT: sigstore-test-364020
2023-04-04T01:43:17.9065142Z   CLOUDSDK_PROJECT: sigstore-test-364020
2023-04-04T01:43:17.9065447Z   GCLOUD_PROJECT: sigstore-test-364020
2023-04-04T01:43:17.9065743Z   GCP_PROJECT: sigstore-test-364020
2023-04-04T01:43:17.9066099Z   GOOGLE_CLOUD_PROJECT: sigstore-test-364020
2023-04-04T01:43:17.9066460Z   CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
2023-04-04T01:43:17.9066799Z   CLOUDSDK_METRICS_ENVIRONMENT_VERSION: 1.1.0
2023-04-04T01:43:17.9067094Z ##[endgroup]
2023-04-04T01:43:19.4059057Z ERROR: (gcloud.auth.print-identity-token) No identity token can be obtained from the current credentials.
2023-04-04T01:43:19.5156942Z ##[error]Process completed with exit code 1.
2023-04-04T01:43:19.5228167Z Post job cleanup.
2023-04-04T01:43:19.6127488Z Removed exported credentials at "/home/runner/work/identity-token-test/identity-token-test/gha-creds-9e65c99f90011dc2.json".
2023-04-04T01:43:19.6227150Z Post job cleanup.
2023-04-04T01:43:19.7953693Z [command]/usr/bin/git version
2023-04-04T01:43:19.8013918Z git version 2.40.0
2023-04-04T01:43:19.8090471Z Temporarily overriding HOME='/home/runner/work/_temp/60c810f8-1f3e-4fbe-80f3-ff9231a40cc4' before making global git config changes
2023-04-04T01:43:19.8094832Z Adding repository directory to the temporary git global config as a safe directory
2023-04-04T01:43:19.8104077Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/identity-token-test/identity-token-test
2023-04-04T01:43:19.8157887Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-04-04T01:43:19.8219631Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-04-04T01:43:19.8483799Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-04-04T01:43:19.8527036Z http.https://github.com/.extraheader
2023-04-04T01:43:19.8545030Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2023-04-04T01:43:19.8593372Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-04-04T01:43:19.9099267Z Cleaning up orphan processes
```

### Additional information

Wasn't sure if this belonged here or at https://github.com/google-github-actions/setup-gcloud or elsewhere, or if this is a known restriction. Feel free to redirect! Thanks!

**Comment by di** [comment]
Thanks for the pointer! I missed that the action can generate ID tokens directly. For any future visitors, this is where the docs describe this: https://github.com/google-github-actions/auth#generating-an-id-token-jwt

**Comment by glasnt** [comment]
Future visitors: this documentation has been moved to https://github.com/google-github-actions/auth/blob/main/docs/EXAMPLES.md#generating-an-id-token-jwt
